### PR TITLE
Fixes pr 1

### PR DESCRIPTION
**Pull request by meguiraun**
only fixes.

Leaving out: dc link, save 2d centring, prepare beamline, will come later

### CODE REVIEW

File: mxcube3/routes/qutils.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 import redis
 import itertools
 import xml.etree.ElementTree as et
-
+import logging
```
Comment (marcus-oscarsson) [inline mxcube3/routes/qutils.py]: logging is never used in this file

File: mxcube3/__init__.py [hunk 2]
```diff
@@ -153,7 +153,11 @@ def complete_initialization(app):
         app.AUTO_MOUNT_SAMPLE = False
         app.AUTO_ADD_DIFFPLAN = False
         app.CENTRING_METHOD = CENTRING_METHOD.LOOP
-        app.NUM_SNAPSHOTS = 4
+        num_snapshots = app.collect.getProperty('num_snapshots')
+        if num_snapshots is not None:
```
Comment (marcus-oscarsson) [inline mxcube3/__init__.py]: I think you can simply do

num_snapshots = app.collect.getProperty('num_snapshots', 4) 

to avoid the if ...

**Comment by marcus-oscarsson** [comment]
Hey 

Thank for all the good fixes, looks good in general.

Please have a look at my comments, and is it possible to do something about the last merge commit ?

Marcus

File: mxcube3/routes/qutils.py (hunk 1 above)
Comment (meguiraun) [reply mxcube3/routes/qutils.py]: it was before I merge with master, removed

**Comment by meguiraun** [comment]
About the merge commit, I already did a rebase, cherry-pick, split ... we have many of those already in master... unless you tell me an easy way to do

File: mxcube3/__init__.py (hunk 2 above)
Comment (meguiraun) [reply mxcube3/__init__.py]: thanks

**Comment by marcus-oscarsson** [comment]
What is upstream for you ?

**Comment by meguiraun** [comment]
thanks

**Comment by meguiraun** [comment]
Upstream is the master repo in github

**Pull request merged**